File: lib/prompts/templates/checkIn.ts
import type { Blueprint, GenerateRequest, SectionOutput, SectionSpec } from '../../types'

export function buildPrompt(
  spec: SectionSpec,
  blueprint: Blueprint,
  req: GenerateRequest,
  priorOutputs?: SectionOutput[]
): string {
  const { lesson_metadata } = req

  const readingPassageOutput = priorOutputs?.find(
    (o) => o.section_type === 'reading_passage'
  )
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  const passageContent = readingPassageOutput?.content as any
  const passageBody: string = passageContent?.passage_body ?? '[Reading passage not yet generated]'
  const passageTitle: string = passageContent?.passage_title ?? 'Reading Passage'

  return `Generate a check-in question for this reading passage.

## Blueprint Context
- Lesson Title: ${blueprint.title}
- Core Concept: ${blueprint.core_concept}
- Essential Question: ${blueprint.essential_question}
- Subject: ${lesson_metadata.subject}
- Grade Level: ${lesson_metadata.grade_level}

## Reading Passage: "${passageTitle}"
${passageBody}

## Instructions
Create ONE open-ended check-in question that:
1. Requires students to find and use evidence from the passage above
2. Connects back to the essential question: "${blueprint.essential_question}"
3. Is thought-provoking but answerable by a Grade ${lesson_metadata.grade_level} student
4. Ends with a directive to use evidence (e.g. "Use evidence from the text to support your answer.")

Also provide a sentence starter that helps students begin their response.

## Output Schema
{
  "question": "string (open-ended, requires evidence from the passage)",
  "sentence_starter": "string (e.g. 'Based on the passage, I think... because...')"
}

Section spec: ${JSON.stringify(spec)}`
}

File: backend/study-guide-agent/app/prompts/templates/check_in.py
from __future__ import annotations

from typing import Any

from app.types import Blueprint, GenerateRequest


def build_prompt(
    spec: dict[str, Any], blueprint: Blueprint, request: GenerateRequest
) -> str:
    passage_title = spec.get("title", "Model Passage")
    passage_text = "\n".join(spec.get("passage", []))
    evidence_focus = spec.get("evidence_focus", "")
    text_features = "\n".join(f"- {item}" for item in spec.get("text_features", []))

    check_in_schema = """{
    "title": "Check-In",
    "passage_title": "string",
    "questions": [
        {
            "number": 1,
            "question": "string",
            "evidence_hint": "string",
            "expected_response_type": "string"
        }
    ]
}"""

    prompt_lines = [
        "Create the check-in section for a K-12 study guide.",
        f"- Lesson title: {blueprint.title}",
        f"- Essential question: {blueprint.essential_question}",
        f"- Source passage title: {passage_title}",
        f"- Evidence focus: {evidence_focus}",
        "- Source passage text:",
        passage_text,
        "- Source text features:",
        text_features or "- none provided",
        "Requirements:",
        "- Write 3 short questions that require evidence from the provided passage text.",
        "- Make each evidence_hint point students toward a specific clue or phrase in the passage.",
        "- Keep the questions aligned to the lesson's core concept and author-purpose analysis.",
        f"- Keep the reading demand close to Grade {request.lesson_metadata.grade_level}.",
        "Expected JSON schema:",
        check_in_schema,
        "Return only JSON.",
    ]

    return "\n".join(prompt_lines)
